### PR TITLE
Add clRNG

### PR DESCRIPTION
**Pull request by jakirkham**
Fixes https://github.com/conda-forge/staged-recipes/issues/3187

Adds a package for clRNG. This builds heavily off of the build strategy used by PyOpenCL and clBLAS.

Would appreciate feedback from interested/knowledgeable parties. Also would welcome interested parties as maintainers.

Note: Some of the build features won't come into play until the feedstock.

cc @inducer

### CODE REVIEW

**Comment by inducer** [comment]
[Random123](http://www.deshawresearch.com/resources_random123.html) is a fantastic CL RNG library. (I've included it in PyOpenCL.)

**Comment by jakirkham** [comment]
Adding a Random123 package would be a good idea too. Looks like it is header only. Does it have any compiler constraints like needing C++11 or similar?

**Comment by jakirkham** [comment]
Actually looks like clRNG has vendored Random123 for [internal]( https://github.com/clMathLibraries/clRNG/tree/v1.0.0-beta/src/cl/include/private/Random123 ) use.

File: recipes/clrng/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+{% if OSX_VARIANT != "native" %}
+  skip: True  # [not osx]
+{% else %}
+  #############################################################
+  # Skip Visual Studio 2015 32-bit builds due to a bug.       #
+  #                                                           #
+  # xref: https://github.com/clMathLibraries/clRNG/issues/11  #
+  #############################################################
+  skip: True  # [(win and py35) or (win32 and py36)]
```
Comment (jakirkham) [inline recipes/clrng/meta.yaml]: Skipping Python 3.5 as Python 3.5 and Python 3.6 use the same VC runtime version. So either will work for our purposes here. This is typical.

What is worth noting is that VS 2015 32-bit builds fail. This [log]( https://ci.appveyor.com/project/conda-forge/staged-recipes/build/1.0.14805/job/rlmkrvh1dr42rf20#L436 ) demonstrates the build failure. An issue has already been raised about this upstream. ( https://github.com/clMathLibraries/clRNG/issues/11 ) As there doesn't seem to be a patch we can use to fix this, we merely disable those builds. VS 2015 64-bit builds seem to build fine.

**Pull request merged**